canonical-web-and-design/maas-ui
# feat(ui): add actions/reducers/selectors for setting an active pod

## Done

- Added action/selectors/reducers for setting active pods.
- Updated pod reducer tests to use the test factories.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Navigate to a KVM details page and check that `pod.active` is set to the pod's id.
- Edit the KVM configuration and check that the pod does not lose the `attached_vlans` parameter in state.
- Navigate back to the KVM list and check that `pod.active` is set back to `null`.

## Fixes

Fixes #1826
